src/cluster_legacy.c

```diff
@@ -5439,6 +5470,7 @@ void clusterCron(void) {
     server.cluster->stats_pfail_nodes = 0;
     /* Run through some of the operations we want to do on each cluster node. */
     di = dictGetSafeIterator(server.cluster->nodes);
+    const long long max_conn_attempts = maxConnectionAttemptsPerCron(dictSize(server.cluster->nodes), server.cluster_node_timeout);
```

Comment:
Should this be server.cluster_node_timeout/2, we try to disconnect and reconnect at half the period of node timeout.

```suggestion
    const long long max_conn_attempts = maxConnectionAttemptsPerCron(dictSize(server.cluster->nodes), server.cluster_node_timeout/2);
```